addons pages won't load (404)

### Describe the issue you are experiencing

Cannot load addon webui's or any config pages after the latest supervisor update.  I use an HA Blue, but accessing from a Unbuntu laptop with Chrome. Mobile app seems to still load ok.

### What is the used version of the Supervisor?

2021-08-1

### What type of installation are you running?

Home Assistant OS

### Which operating system are you running on?

Home Assistant Operating System

### What is the version of your installed operating system?

6.2

### What version of Home Assistant Core is installed?

2021.9.0b6

### Steps to reproduce the issue

1. Go into supervisor dashboard.
2.Try to get to addon config, eg Node Red, InfluDB etc
3.
...


### Anything in the Supervisor logs that might be useful for us?

```txt
21-08-31 00:51:51 INFO (MainThread) [supervisor.resolution.checks.base] Run check for IssueType.PWNED/ContextType.ADDON
21-08-31 00:51:51 WARNING (MainThread) [supervisor.addons.options] Option 'interface' does not exist in the schema for Samba share (core_samba)
21-08-31 00:51:51 WARNING (MainThread) [supervisor.addons.options] Option 'anonymous' does not exist in the schema for Mosquitto broker (core_mosquitto)
21-08-31 00:51:51 WARNING (MainThread) [supervisor.addons.options] Option 'dark_mode' does not exist in the schema for Node-RED (a0d7b954_nodered)
21-08-31 00:51:51 WARNING (MainThread) [supervisor.addons.options] Option 'require_ssl' does not exist in the schema for Node-RED (a0d7b954_nodered)
21-08-31 00:51:51 WARNING (MainThread) [supervisor.addons.options] Unknown option 'new_api' for Zigbee2mqtt (7ad98f9c_zigbee2mqtt)
21-08-31 00:51:52 INFO (MainThread) [supervisor.resolution.checks.base] Run check for IssueType.SECURITY/ContextType.CORE
21-08-31 00:51:52 INFO (MainThread) [supervisor.resolution.checks.base] Run check for IssueType.TRUST/ContextType.PLUGIN
21-08-31 00:51:54 INFO (MainThread) [supervisor.api.middleware.security] /snapshots access from cebe7a76_hassio_google_drive_backup
21-08-31 00:51:55 WARNING (MainThread) [supervisor.dbus.network] Can't process /org/freedesktop/NetworkManager/Devices/41: Error: GDBus.Error:org.freedesktop.DBus.Error.UnknownMethod: No such interface “org.freedesktop.DBus.Properties” on object at path /org/freedesktop/NetworkManager/Devices/41
```


### Additional information

Chrome console log (seems to be a 404)

entrypoint.7ca7c83a.js:5527 GET http://homeassistant.local:8123/api/hassio/app/frontend_latest/f2ea8216.js net::ERR_ABORTED 404 (Not Found)
s.l @ entrypoint.7ca7c83a.js:5527
s.f.j @ entrypoint.7ca7c83a.js:5527
(anonymous) @ entrypoint.7ca7c83a.js:5527
s.e @ entrypoint.7ca7c83a.js:5527
load @ entrypoint.7ca7c83a.js:2734
value @ entrypoint.7ca7c83a.js:5032
performUpdate @ entrypoint.7ca7c83a.js:2794
Πq @ entrypoint.7ca7c83a.js:2794
async function (async)
Πq @ entrypoint.7ca7c83a.js:2794
requestUpdate @ entrypoint.7ca7c83a.js:2794
set @ entrypoint.7ca7c83a.js:2794
W @ entrypoint.7ca7c83a.js:4181
I @ entrypoint.7ca7c83a.js:4181
v @ entrypoint.7ca7c83a.js:4181
_ @ entrypoint.7ca7c83a.js:4181
I @ entrypoint.7ca7c83a.js:4181
O @ entrypoint.7ca7c83a.js:4181
update @ entrypoint.7ca7c83a.js:4181
performUpdate @ entrypoint.7ca7c83a.js:2794
Πq @ entrypoint.7ca7c83a.js:2794
async function (async)
Πq @ entrypoint.7ca7c83a.js:2794
requestUpdate @ entrypoint.7ca7c83a.js:2794
set @ entrypoint.7ca7c83a.js:2794
(anonymous) @ custom-panel.b37826f4.js:1348
(anonymous) @ custom-panel.b37826f4.js:1348
p @ custom-panel.b37826f4.js:1348
value @ 6da8e7d0.js:1
performUpdate @ app.f08a72d3.js:1
_$EC @ app.f08a72d3.js:1
async function (async)
_$EC @ app.f08a72d3.js:1
requestUpdate @ app.f08a72d3.js:1
set @ app.f08a72d3.js:1
value @ app.f08a72d3.js:35925
value @ app.f08a72d3.js:29560
performUpdate @ app.f08a72d3.js:1
_$EC @ app.f08a72d3.js:1
async function (async)
_$EC @ app.f08a72d3.js:1
requestUpdate @ app.f08a72d3.js:1
set @ app.f08a72d3.js:1
P @ app.f08a72d3.js:5264
_$AI @ app.f08a72d3.js:5264
m @ app.f08a72d3.js:5264
T @ app.f08a72d3.js:5264
_$AI @ app.f08a72d3.js:5264
O @ app.f08a72d3.js:5264
update @ app.f08a72d3.js:5264
performUpdate @ app.f08a72d3.js:1
_$EC @ app.f08a72d3.js:1
async function (async)
_$EC @ app.f08a72d3.js:1
requestUpdate @ app.f08a72d3.js:1
set @ app.f08a72d3.js:1
P @ app.f08a72d3.js:5264
_$AI @ app.f08a72d3.js:5264
m @ app.f08a72d3.js:5264
T @ app.f08a72d3.js:5264
_$AI @ app.f08a72d3.js:5264
O @ app.f08a72d3.js:5264
update @ app.f08a72d3.js:5264
performUpdate @ app.f08a72d3.js:1
_$EC @ app.f08a72d3.js:1
async function (async)
_$EC @ app.f08a72d3.js:1
requestUpdate @ app.f08a72d3.js:1
set @ app.f08a72d3.js:1
t @ app.f08a72d3.js:36741
(anonymous) @ app.f08a72d3.js:36741
r @ entrypoint.7ca7c83a.js:4181
n @ entrypoint.7ca7c83a.js:4181
value @ entrypoint.7ca7c83a.js:1177
handleEvent @ entrypoint.7ca7c83a.js:4181
entrypoint.7ca7c83a.js:5032 Error loading page addon ChunkLoadError: Loading chunk 841 failed.
(error: http://homeassistant.local:8123/api/hassio/app/frontend_latest/f2ea8216.js)
    at Object.s.f.j (entrypoint.7ca7c83a.js:5527)
    at entrypoint.7ca7c83a.js:5527
    at Array.reduce (<anonymous>)
    at Function.s.e (entrypoint.7ca7c83a.js:5527)
    at Object.load (entrypoint.7ca7c83a.js:2734)
    at HTMLElement.value (entrypoint.7ca7c83a.js:5032)
    at HTMLElement.performUpdate (entrypoint.7ca7c83a.js:2794)
    at HTMLElement.Πq (entrypoint.7ca7c83a.js:2794)

--- a/homeassistant/components/hassio/http.py
+++ b/homeassistant/components/hassio/http.py
@@ -10,6 +10,7 @@ import aiohttp
 from aiohttp import web
 from aiohttp.client import ClientTimeout
 from aiohttp.hdrs import (
+    CACHE_CONTROL,
     CONTENT_ENCODING,
     CONTENT_LENGTH,
     CONTENT_TYPE,
@@ -50,6 +51,8 @@ NO_AUTH_ONBOARDING = re.compile(
 NO_AUTH = re.compile(
     r"^(?:" r"|app/.*" r"|addons/[^/]+/logo" r"|addons/[^/]+/icon" r")$"
 )
+
+NO_STORE = re.compile(r"^(?:" r"|app/entrypoint.js" r")$")
 
 
 class HassIOView(HomeAssistantView):
@@ -104,7 +107,7 @@ class HassIOView(HomeAssistantView):
 
             # Stream response
             response = web.StreamResponse(
-                status=client.status, headers=_response_header(client)
+                status=client.status, headers=_response_header(client, path)
             )
             response.content_type = client.content_type
 
@@ -139,7 +142,7 @@ def _init_header(request: web.Request) -> dict[str, str]:
     return headers
 
 
-def _response_header(response: aiohttp.ClientResponse) -> dict[str, str]:
+def _response_header(response: aiohttp.ClientResponse, path: str) -> dict[str, str]:
     """Create response header."""
     headers = {}
 
@@ -152,6 +155,9 @@ def _response_header(response: aiohttp.ClientResponse) -> dict[str, str]:
         ):
             continue
         headers[name] = value
+
+    if NO_STORE.match(path):
+        headers[CACHE_CONTROL] = "no-store, max-age=0"
 
     return headers
 

--- a/tests/components/hassio/test_http.py
+++ b/tests/components/hassio/test_http.py
@@ -185,3 +185,21 @@ async def test_stream(hassio_client, aioclient_mock):
     aioclient_mock.get("http://127.0.0.1/test")
     await hassio_client.get("/api/hassio/test", data="test")
     assert isinstance(aioclient_mock.mock_calls[-1][2], StreamReader)
+
+
+async def test_entrypoint_cache_control(hassio_client, aioclient_mock):
+    """Test that we return cache control for requests to the entrypoint only."""
+    aioclient_mock.get("http://127.0.0.1/app/entrypoint.js")
+    aioclient_mock.get("http://127.0.0.1/app/entrypoint.fdhkusd8y43r.js")
+
+    resp1 = await hassio_client.get("/api/hassio/app/entrypoint.js")
+    resp2 = await hassio_client.get("/api/hassio/app/entrypoint.fdhkusd8y43r.js")
+
+    # Check we got right response
+    assert resp1.status == 200
+    assert resp2.status == 200
+
+    assert len(aioclient_mock.mock_calls) == 2
+    assert resp1.headers["Cache-Control"] == "no-store, max-age=0"
+
+    assert "Cache-Control" not in resp2.headers

